External/FEXCore/include/FEXCore/Core/CoreState.h

```diff
@@ -14,7 +14,9 @@ namespace FEXCore::Core {
     uint64_t rip; ///< Current core's RIP. May not be entirely accurate while JIT is active
     uint64_t gregs[16];
     uint64_t : 64;
-    uint64_t xmm[16][2];
+    uint64_t : 64;
+    uint64_t : 64;
+    uint64_t xmm[16][4];
     uint16_t es, cs, ss, ds;
```

Comment:
We could move 
```
    uint16_t es, cs, ss, ds;
    uint64_t gs;
    uint64_t fs;
```
info the padding

Reply:
When I move these into the padding the tests start breaking. Are we hardcoding offsets somewhere other than DeadContextStoreElimination pass that need to be accounted for? (latest commit is what I've done so far)

Reply:
```
UNREACHABLE in /mnt/Work/Work/work/FEXNew/External/vixl/src/aarch64/assembler-aarch64.cc, line 5876

Program received signal SIGABRT, Aborted.
__pthread_kill_implementation (threadid=549621534752, signo=signo@entry=6, no_tid=no_tid@entry=0) at ./nptl/pthread_kill.c:44
44      ./nptl/pthread_kill.c: No such file or directory.
(gdb) bt
#0  __pthread_kill_implementation (threadid=549621534752, signo=signo@entry=6, no_tid=no_tid@entry=0) at ./nptl/pthread_kill.c:44
#1  0x0000007ff7b6f254 in __pthread_kill_internal (signo=6, threadid=<optimised out>) at ./nptl/pthread_kill.c:78
#2  0x0000007ff7b2a67c in __GI_raise (sig=sig@entry=6) at ../sysdeps/posix/raise.c:26
#3  0x0000007ff7b17130 in __GI_abort () at ./stdlib/abort.c:79
#4  0x0000005555c61b60 in vixl::aarch64::Assembler::LoadStoreMemOperand (this=0x7ff76b7040, addr=..., access_size_in_bytes_log2=4, option=vixl::aarch64::PreferUnscaledOffset) at /mnt/Work/Work/work/FEXNew/External/vixl/src/aarch64/assembler-aarch64.cc:5876
#5  0x0000005555c205d0 in vixl::aarch64::Assembler::LoadStore (this=0x7ff76b7040, rt=..., addr=..., op=vixl::aarch64::STR_q, option=vixl::aarch64::PreferUnscaledOffset) at /mnt/Work/Work/work/FEXNew/External/vixl/src/aarch64/assembler-aarch64.cc:5886
#6  0x0000005555c21610 in vixl::aarch64::Assembler::stur (this=0x7ff76b7040, rt=..., dst=..., option=vixl::aarch64::PreferUnscaledOffset) at /mnt/Work/Work/work/FEXNew/External/vixl/src/aarch64/assembler-aarch64.cc:1338
#7  0x00000055559994d4 in FEXCore::CPU::Arm64JITCore::Op_StoreContextIndexed (this=0x7ff76b7000, IROp=0x7ff761e6e5, Node=...) at /mnt/Work/Work/work/FEXNew/External/FEXCore/Source/Interface/Core/JIT/Arm64/MemoryOps.cpp:428
#8  0x0000005555945e1c in FEXCore::CPU::Arm64JITCore::CompileCode (this=0x7ff76b7000, Entry=65539, IR=0x7ff771e530, DebugData=0x7ff76211c0, RAData=0x7ff771e500, GDBEnabled=false) at /mnt/Work/Work/work/FEXNew/External/FEXCore/Source/Interface/Core/JIT/Arm64/JIT.cpp:758
#9  0x0000005555741768 in FEXCore::Context::Context::CompileCode (this=0x7ff769e000, Thread=0x7ff76ad000, GuestRIP=65539) at /mnt/Work/Work/work/FEXNew/External/FEXCore/Source/Interface/Core/Core.cpp:974
#10 0x0000005555741bfc in FEXCore::Context::Context::CompileBlock (this=0x7ff769e000, Frame=0x7ff76ad1b0, GuestRIP=65539) at /mnt/Work/Work/work/FEXNew/External/FEXCore/Source/Interface/Core/Core.cpp:1014
#11 0x0000005555741aec in FEXCore::Context::Context::CompileBlockJit (this=0x7ff769e000, Frame=0x7ff76ad1b0, GuestRIP=65539) at /mnt/Work/Work/work/FEXNew/External/FEXCore/Source/Interface/Core/Core.cpp:985
#12 0x0000007ff7fc1334 in ?? ()
```
Looks like an alignment issue. mm members need to be 128-bit aligned and we didn't have a static_assert for it.

Reply:
Removing padding and adjusting the RCLSE classification resolves the problems here.

Reply:
Thanks. Added a static assert as well so that this doesn't slip through again